### ISSUE TITLE
golang: Illegal Instruction when arm cpu do not have VFP

### ISSUE THREAD

**Issue by kuoruan** [openwrt/packages#10967]
Maintainer: @jefferyto
Environment: armv7

Description:

Some CPU do not have VFP.

For example: Phicomm K3

the default GOARM for K3 is 7, but the build would cause Illegal Instruction. (golang/go#18483)

to make it work, we need to set GOARM to 5.

```
root@Phicomm-K3:~# cat /proc/cpuinfo
processor : 0
model name : ARMv7 Processor rev 0 (v7l)
BogoMIPS : 1400.00
Features : half thumb fastmult edsp tls
CPU implementer : 0x41
CPU architecture: 7
CPU variant : 0x3
CPU part : 0xc09
CPU revision : 0

processor : 1
model name : ARMv7 Processor rev 0 (v7l)
BogoMIPS : 1400.00
Features : half thumb fastmult edsp tls
CPU implementer : 0x41
CPU architecture: 7
CPU variant : 0x3
CPU part : 0xc09
CPU revision : 0

Hardware : BCM5301X
Revision : 0000
Serial : 0000000000000000
```

**Comment by jefferyto** [comment]
Thanks for reporting this - I think I've figured out how to address this, but I will wait until #10971 is done to open a PR for this fix.

**Comment by jefferyto** [comment]
Fixes are in #11020 (master), #11021 (19.07), and #11023 (18.06)

**Comment by kuoruan** [comment]
Great! Thank you.
